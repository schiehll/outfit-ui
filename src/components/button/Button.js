import React, {PropTypes} from 'react'
import suitup from 'suit-up'
import style from './Button.style'
import colors from '../../config/colors'
import sizes from '../../config/sizes'
import types from '../../config/types'
import removeInvalidProps from '../../utils/removeInvalidProps'

const Button = props => {
  const {children, className, styles, ...otherProps} = props

  let classes = ``
  let shouldBeDefault = true

  const possibleProps = colors.concat(sizes).concat(types)
  possibleProps.forEach(prop => {
    if (props[prop]) {
      if (shouldBeDefault &&
        prop !== 'default' &&
        colors.indexOf(prop) >= 0) {
        shouldBeDefault = false
      }
      classes += ` ${styles[prop]} `
    }
  })

  if (shouldBeDefault) classes += `${styles.default}`
  if (className) classes += ` ${className}`

  const validProps = removeInvalidProps(otherProps)

  return (
    <button
      className={classes.trim()}
      {...validProps}
    >
      {children}
    </button>
  )
}

Button.propTypes = {
  className: PropTypes.string,
  outline: PropTypes.bool,
  round: PropTypes.bool,
  block: PropTypes.bool,
  small: PropTypes.bool,
  big: PropTypes.bool,
  primary: PropTypes.bool,
  success: PropTypes.bool,
  warning: PropTypes.bool,
  danger: PropTypes.bool
}

export default suitup(style)(Button)
