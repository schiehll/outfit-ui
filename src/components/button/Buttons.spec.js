import React from 'react'
import {render} from 'enzyme'
import {renderToJson} from 'enzyme-to-json'
import {ThemeProvider} from 'suit-up'
import Button from './index'

let styles
let theme

beforeAll(() => {
  theme = require('../../theme.example').default

  styles = {
    default: 'default',
    primary: 'primary',
    success: 'success',
    info: 'info',
    warning: 'warning',
    danger: 'danger',
    round: 'round',
    outline: 'outline'
  }
})

test('should render just fine', () => {
  const props = {styles}

  const component = render(
    <ThemeProvider theme={theme}>
      <Button {...props}>Btn</Button>
    </ThemeProvider>
  )

  expect(renderToJson(component)).toMatchSnapshot()
})

test('should accept props', () => {
  let props = {styles, primary: true}
  let component = render(
    <ThemeProvider theme={theme}>
      <Button {...props}>Btn</Button>
    </ThemeProvider>
  )

  expect(renderToJson(component)).toMatchSnapshot()

  props = {styles, success: true}
  component = render(
    <ThemeProvider theme={theme}>
      <Button {...props}>Btn</Button>
    </ThemeProvider>
  )

  expect(renderToJson(component)).toMatchSnapshot()

  props = {styles, info: true}
  component = render(
    <ThemeProvider theme={theme}>
      <Button {...props}>Btn</Button>
    </ThemeProvider>
  )

  expect(renderToJson(component)).toMatchSnapshot()

  props = {styles, round: true, outline: true}
  component = render(
    <ThemeProvider theme={theme}>
      <Button {...props}>Btn</Button>
    </ThemeProvider>
  )

  expect(renderToJson(component)).toMatchSnapshot()

  props = {styles, disabled: true}
  component = render(
    <ThemeProvider theme={theme}>
      <Button {...props}>Btn</Button>
    </ThemeProvider>
  )

  expect(renderToJson(component)).toMatchSnapshot()
})

test('should accept a className and merge it', () => {
  const props = {styles}

  const component = render(
    <ThemeProvider theme={theme}>
      <Button className='someClass' {...props}>Btn</Button>
    </ThemeProvider>
  )

  expect(renderToJson(component)).toMatchSnapshot()
})
