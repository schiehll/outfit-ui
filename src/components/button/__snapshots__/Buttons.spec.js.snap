exports[`test should accept a className and merge it 1`] = `
<button
  class="default_1hm0j base_1hm0j someClass">
  Btn
</button>
`;

exports[`test should accept props 1`] = `
<button
  class="primary_1hm0j base_1hm0j">
  Btn
</button>
`;

exports[`test should accept props 2`] = `
<button
  class="success_1hm0j base_1hm0j">
  Btn
</button>
`;

exports[`test should accept props 3`] = `
<button
  class="info_1hm0j base_1hm0j">
  Btn
</button>
`;

exports[`test should accept props 4`] = `
<button
  class="outline_1hm0j  round_1hm0j default_1hm0j base_1hm0j">
  Btn
</button>
`;

exports[`test should accept props 5`] = `
<button
  class="default_1hm0j base_1hm0j"
  disabled="">
  Btn
</button>
`;

exports[`test should render just fine 1`] = `
<button
  class="default_1hm0j base_1hm0j">
  Btn
</button>
`;
